tests/functional/CMakeLists.txt

```diff
@@ -15,58 +15,136 @@
+    func_bsplib_example_put_array.cpp
+    func_bsplib_example_put_array_unsafemode.cpp
+    func_bsplib_example_reverse.cpp
+    func_bsplib_example_reverse_unsafemode.cpp
+    func_bsplib_get_exceptions.cpp
+    func_bsplib_get_normal.cpp
+    func_bsplib_get_normal_unsafemode.cpp
+    func_bsplib_get_twice_on_same_remote.cpp
+    func_bsplib_get_twice_on_same_remote_unsafemode.cpp
+    func_bsplib_getput_same_dest.cpp
+    func_bsplib_getput_same_dest_unsafemode.cpp
+    func_bsplib_getput_same_remote.cpp
+    func_bsplib_getput_same_remote_unsafemode.cpp
+    func_bsplib_getput_zero_bytes.cpp
+    func_bsplib_hpget_many.cpp
+    func_bsplib_hpput_many.cpp
+    func_bsplib_hpsend_many.cpp
+    func_bsplib_nprocs.cpp
+    func_bsplib_pid.cpp
+    func_bsplib_pushpopreg_ambiguous.cpp
+    func_bsplib_pushpopreg_different_variables.cpp
+    func_bsplib_pushpopreg_exceptions.cpp
+    func_bsplib_pushpopreg_many_same.cpp
+    func_bsplib_pushpopreg_normal.cpp
+    func_bsplib_pushpopreg_normal_unsafemode.cpp
+    func_bsplib_pushpopreg_null.cpp
+    func_bsplib_pushpopreg_pop_before_put.cpp
+    func_bsplib_pushpopreg_pop_before_put_unsafemode.cpp
+    func_bsplib_pushpopreg_pop_on_one_process.cpp
+    func_bsplib_pushpopreg_push_on_one_process.cpp
+    func_bsplib_pushpopreg_same_growing_memory.cpp
+    func_bsplib_pushpopreg_same_shrinking_memory.cpp
+    func_bsplib_pushpopreg_two_pops_before_two_puts.cpp
+    func_bsplib_put_exceptions.cpp
+    func_bsplib_put_normal.cpp
+    func_bsplib_put_normal_unsafemode.cpp
+    func_bsplib_send_empty_tag.cpp
+    func_bsplib_send_non_empty_tag.cpp
+    func_bsplib_send_none.cpp
+    func_bsplib_send_null.cpp
+    func_bsplib_send_one.cpp
+    func_bsplib_send_one_unsafemode.cpp
+    func_bsplib_set_different_tag_size.cpp
+    func_bsplib_set_tag_size.cpp
+    func_bsplib_sync_except_p0.cpp
+    func_bsplib_sync_only_p0.cpp
+    func_bsplib_time.cpp
+    func_lpf_deregister_parallel_multiple.cpp
+    func_lpf_deregister_parallel_single.cpp
+    func_lpf_exec_multiple_call_single_arg_dual_proc.cpp
+    func_lpf_exec_nested_call_single_arg_dual_proc.cpp
+    func_lpf_exec_single_call_no_arg_max_proc.cpp
+    func_lpf_exec_single_call_no_arg_single_proc.cpp
+    func_lpf_exec_single_call_single_arg_dual_proc.cpp
+    func_lpf_exec_single_call_single_arg_max_proc_early_exit_one.cpp
+    func_lpf_exec_single_call_single_arg_max_proc_early_exit_zero.cpp
+    func_lpf_exec_single_call_single_arg_single_proc.cpp
+    func_lpf_get_parallel_alltoall.cpp
+    func_lpf_get_parallel_huge.cpp
+    func_lpf_get_parallel_overlapping_complete.cpp
+    func_lpf_get_parallel_overlapping_pyramid.cpp
+    func_lpf_get_parallel_overlapping_rooftiling.cpp
+    func_lpf_get_parallel_single.cpp
+    #func_lpf_hook_simple.mpirma.cpp
+    #func_lpf_hook_simple.pthread.cpp
+    #func_lpf_hook_subset.mpimsg.cpp
+    #func_lpf_hook_tcp.mpirma.cpp
+    #func_lpf_hook_tcp_timeout.mpirma.cpp
```

Comment:
These should be enabled right, if (and only if) the selected MPI is not OpenMPI ?

Reply:
I fixed all of these EXCEPT for func_lpf_hook_tcp.mpirma.cpp, which requires some input arguments not documented anywhere